api/api.js

```diff
@@ -1,247 +1,295 @@
+        earliestDueDate: { $min: '$dueDate' }
+      }
+    }],
+    function (err, result) {
+      if (result[0]) {
+        users.update({ username: username },
+          { $set: { earliestDueDate: result[0].earliestDueDate }
+        });
+      } else {
+        users.update({ username: username },
+          { $set: { earliestDueDate: new Date().setYear(3000) }
+        });
+      }
+      callback();
+    });
+};
+
+if (process.env.VCAP_SERVICES) {
   vcapServices = JSON.parse(process.env.VCAP_SERVICES);
-  url = vcapServices["mongodb26-swarm"][0].credentials.uri
+  url = vcapServices['mongodb26-swarm'][0].credentials.uri;
 } else {
   url = 'mongodb://localhost:27017/checklistomania';
 }
 
-MongoClient.connect(url, function(err, db) {
-  module.exports.db = db;
-  items = db.collection("items");
-  checklists = db.collection("checklists");
-  checklists.remove({})
-  users = db.collection("users");
+MongoClient.connect(url, function (connectErr, db) {
+  items = db.collection('items');
+  checklists = db.collection('checklists');
+  users = db.collection('users');
 
+  checklists.remove({});
   items.ensureIndex('owner');
   users.ensureIndex('username');
 
-  fs.readdir('./checklists', function(err, files) {
-    files.forEach(function(fileName) {
-    var filePath = 'checklists/' + fileName;
-    fs.readFile(filePath, 'utf8',
-      function(err, data) {
-        checklist = JSON.parse(data);
+  module.exports.db = db;
+
+  fs.readdir('./checklists', function (readdirErr, files) {
+    files.forEach(function (fileName) {
+      var filePath = 'checklists/' + fileName;
+      fs.readFile(filePath, 'utf8', function (readFileErr, data) {
+        var checklist = JSON.parse(data);
         checklists.insert(checklist);
       });
     });
   });
 });
 
-var router = express.Router();
-
-router.get('/', function(req, res) {
-    res.json({ message: 'hooray! welcome to the api!' });
+router.get('/', function (req, res) {
+  res.json({ message: 'hooray! welcome to the api!' });
 });
 
 router.get('/isalive', function (req, res) {
   res.send('OK');
 });
 
 function addChildrenToChecklist(checklist) {
-  Object.keys(checklist.items).forEach(function(itemId) {checklist.items[itemId].children = []});
-  Object.keys(checklist.items).forEach(function(itemId) {
-    checklist.items[itemId].dependsOn.forEach(function(dependencyItemId) {
-      if(checklist.items[dependencyItemId].children) {
+  Object.keys(checklist.items).forEach(function (itemId) {
+    checklist.items[itemId].children = [];
+  });
+  Object.keys(checklist.items).forEach(function (itemId) {
+    checklist.items[itemId].dependsOn.forEach(function (dependencyItemId) {
+      if (checklist.items[dependencyItemId].children) {
         checklist.items[dependencyItemId].children.push(itemId);
-      };
+      }
     });
   });
 
   return checklist;
 }
 
 function getSortedItemIds(checklist) {
+  var actionable;
+
   checklist = JSON.parse(JSON.stringify(checklist));
```

Comment:
@anthonygarvan does this serve any purpose? it seems like stringifying and parseing should have no effect. Tests still pass when this line is removed completely, which is what I'd expect.